### PR TITLE
[Backend] Add simple request validators

### PR DESCRIPTION
**Pull request by valio7**

### CODE REVIEW

File: backend/src/validators/request-validators/UserRequestValidator.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { User } from "../../domain/User";
+import { EmptyEmailError } from "../../errors/EmptyEmailError";
+import { EmptyPasswordError } from "../../errors/EmptyPasswordError";
+import { EmptyUsernameError } from "../../errors/EmptyUsernameError";
+
+
+export class UserRequestValidator {
+
+    static validateCreateUser(user: User) {
+        if (user.email.trim().length == 0) {
```
Comment (niksan258) [inline backend/src/validators/request-validators/UserRequestValidator.ts:11]: consider validating email using regex

File: backend/src/validators/request-validators/UserRequestValidator.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        this.validateUsernameAndPassword(user.username, user.password);
+    }
+
+    static validateLoginUser(user: ILoginUserDto) {
+        this.validateUsernameAndPassword(user.username, user.password);
+    }
+
+    private static validateUsernameAndPassword(username: string, password: string) {
```
Comment (niksan258) [inline backend/src/validators/request-validators/UserRequestValidator.ts:22]: Maybe add minimum username length & pass requirements (8 chars, uppercase letter, special symbol?)

File: backend/src/validators/request-validators/UserRequestValidator.ts (hunk 1 above)
Comment (valio7) [reply backend/src/validators/request-validators/UserRequestValidator.ts:11]: I am currently working on it but in my opinion it is more like a businesss validation and not a request issue.

File: backend/src/validators/request-validators/UserRequestValidator.ts (hunk 2 above)
Comment (valio7) [reply backend/src/validators/request-validators/UserRequestValidator.ts:22]: Will do in the use case where the business validation is made

**Review by niksan258 — APPROVED** [review]

**Pull request merged**